dev/check_function_signatures.py

```diff
@@ -0,0 +1,210 @@
+import argparse
```

Comment:
This PR is the best effort to prevent accidentally introducing breaking changes.